src/pkg/shub/client/util.go

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2018, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package client
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+// ShubParseReference accepts a Shub reference string and parses its content
+// It will return an error if the given URI is not valid,
+// otherwise it will parse the contents into a ShubURI struct
+func ShubParseReference(src string) (uri ShubURI, err error) {
+
+	//define regex for each URI component
+	registryRegexp := `([-a-zA-Z0-9/]{1,64}\/)?` //target is very open, outside registry
```

Comment:
should registry be able to be a dns name here? so it needs to allow periods?

i.e. `shub://myprivateregistry.sylabs.io/dtrudg/alpine:latest`